src/Strategy/Python/Poetry/Common.hs

```diff
@@ -154,20 +176,29 @@ toCanonicalName :: Text -> Text
 toCanonicalName t = toLower $ replace "_" "-" (replace "." "-" t)
 
 -- | Maps poetry lock package to map of package name and associated dependency.
-toMap :: [PoetryLockPackage] -> Map.Map PackageName Dependency
-toMap pkgs = Map.fromList $ (\x -> (canonicalPkgName x, toDependency x)) <$> (filter supportedPoetryLockDep pkgs)
+toMap :: [PackageName] -> [PoetryLockPackage] -> Map.Map PackageName Dependency
+toMap prodPkgs pkgs = Map.fromList $ (\x -> (canonicalPkgName x, toDependency x)) <$> (filter supportedPoetryLockDep pkgs)
```

Comment:
Could this be renamed to something like `makePackageToLockDependencyMap`? I know you didn't write this but `toMap` tells me very little about the purpose of this function at call-sites.